cupyx/scipy/ndimage/_morphology.py

```diff
@@ -537,7 +539,8 @@ def binary_closing(input, structure=None, iterations=1, output=None, origin=0,
 
     .. seealso:: :func:`scipy.ndimage.binary_closing`
     """
-    structure, _, _ = _prep_structure(structure, input.ndim)
+    axes = _util._check_axes(axes, input.ndim)
+    structure, _, _ = _prep_structure(structure, len(axes))
```

Comment:
```suggestion
    structure, _, _ = _prep_structure(structure, input.ndim)
```

Reply:
All of these should stay as is. I had missed one case of changing `input.ndim -> len(axes)` in `binary_hit_or_miss` (now fixed in bf2647c)

Reply:
Let's keep this conversation in one thread for simplicity

Given the discussion that has already happened above, let's stick with this one: https://github.com/cupy/cupy/pull/8954#discussion_r1969297034

Please take a look at the question there